pyannote/pipeline/pipeline.py

```diff
@@ -165,21 +166,30 @@ def _flattened_parameters(
             Flattened dictionary of parameters.
         """
 
+
+
         if frozen and instantiated:
             msg = "one must choose between `frozen` and `instantiated`."
             raise ValueError(msg)
 
         # initialize dictionary with root parameters
         if instantiated:
-            params = dict(self._instantiated)
+            params = dict(flatten_structured(self._instantiated))
 
         elif frozen:
             params = {
                 n: p.value for n, p in self._parameters.items() if isinstance(p, Frozen)
             }
+            params = dict(flatten_structured(params))
```

Comment:
What if only part of the structured parameters are frozen? Does this work? 
Or do I understand this correctly that either the whole structured parameters is frozen or it's not.

Reply:
ah, this is tricky. In _theory_, you could very well freeze subparameters, although in this case, it'll consider only the frozen "root" parameter. This could be fixed, but the freezing logic probably should be overhauled as well. My take on this issue was that "people" are mostly going to freeze the whole structure when needed.

Reply:
An afterthought on this issue: what would be needed to have a working freezing capability even for structured parameters:
- make the `_flatten` method able to search the structured parameters for frozen subparameters
- add a `freeze` method to structured parameters, so users can do something like `my_pipeline.parameters().param_dict["param_a"].freeze(value)`